main.py

```diff
@@ -0,0 +1,162 @@
+        # https://cloud.google.com/functions/docs/monitoring/logging#writing_structured_logs
+        raise InvalidRequestError("Method not allowed", 405)
+
+    if not request.json:
+        raise InvalidRequestError("Missing notification request data", 422)
+
+    data = request.json["payload"]["fulfilmentRequest"]
+
+    log_context = {
+        "tx_id": data.get("tx_id"),
+        "questionnaire_id": data.get("questionnaire_id"),
+    }
+
+    required_keys = ("form_type", "region_code", "language_code")
+    if missing_keys := [key for key in required_keys if not data.get(key)]:
+        msg = f"Missing {', '.join(missing_keys)} identifier(s)"
+        raise InvalidRequestError(msg, 422, log_context)
+
+    try:
+        template_id = template_id_mapping[
+            (data["form_type"], data["region_code"], data["language_code"])
+        ]
+    except KeyError as exc:
+        raise InvalidRequestError("No template id selected", 422, log_context) from exc
+
+    return {key: data.get(key) for key in data_fields}, template_id, log_context
+
+
+# pylint: disable=too-many-return-statements
+def send_email(request: Request) -> Tuple[str, int]:
+    try:
+        data, template_id, log_context = _validate_request(request)
+    except InvalidRequestError as error:
+        log_context = error.log_context if error.log_context else {}
+        log_error(message=error.message, **log_context)
+        return error.message, error.status_code
+
+    api_token = create_jwt_token(secret_key, service_id)
+    session.headers.update(
+        {
+            "Content-type": "application/json",
+            "Authorization": f"Bearer {api_token}",
+        }
+    )
+
+    kwargs = json.dumps(
+        {
+            "template_id": template_id,
+            "personalisation": {"address": data["display_address"]},
+            "email_address": data["email_address"],
+        }
+    )
+
+    try:
+        response = session.post(
+            f"{NOTIFY_BASE_URL}/notifications/email",
+            kwargs,
+        )
+        response.raise_for_status()
+        entry = dict(**log_context, message="notify email requested")
+        log_info(**entry)
+    except RequestException as error:
+        error_message = error.response.json()["errors"][0]
+        status_code = error.response.status_code
+        log_error(
+            **log_context,
+            status_code=status_code,
+            message=f"notify request failed: {error_message}",
+        )
+        return "Notify request failed", error.response.status_code
+
+    if response.status_code == 204:
+        return "No content", 204
+
+    try:
+        response_content = response.json()
+        del response_content["content"]
+        log_info(**log_context, **response_content)
+        return "Notify request successful", response.status_code
+    except ValueError:
+        message = "Notify JSON response object failed decoding"
+        log_error(**log_context, message=message)
```

Comment:
If `message` was a required first param:
```suggestion
        log_error("Notify JSON response object failed decoding", **log_context)
```